Vector.c

```diff
@@ -44,10 +46,15 @@ void Vector_delete(Vector* this) {
    free(this);
 }
 
+static inline bool Vector_isDirty(const Vector* this) {
+   return this->dirty_count > 0;
+}
+
 #ifndef NDEBUG
 
 static bool Vector_isConsistent(const Vector* this) {
    assert(this->items <= this->arraySize);
+   assert(Vector_isDirty(this) == false);
```

Comment:
```suggestion
   assert(!Vector_isDirty(this));
```

You don't say "if the weather is nice is true" either …